### DISCUSSION TITLE
Is there a way to lower data usage in CDP mode?

### DISCUSSION THREAD

**Discussion by krzysztofkub** [seleniumbase/SeleniumBase#3339] [General]
Hi, I can see that running CDP mode uses a lot of proxy data. Is there a way to use some caching logic to make it lower?

**Comment by mdmintz** [comment]
Need more info to see what you're doing.
There are [different syntax formats](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md).

**Comment by krzysztofkub** [comment]
Ok, I'll try to explain without confusing too much

```
while True:    
    with SB(
            uc=True,
            incognito=True,
            locale_code="en",
            headed=True,
            xvfb=True,
            xvfb_metrics="1920,1080",
            proxy=proxy,
    ) as sb:

        url = "XXXX"
        sb.activate_cdp_mode(url)
        pyautogui._pyautogui_x11._display = Xlib.display.Display(display)
        #do some clicks etc.
        sb.driver.quit()
```
        
 I'm running this code in endless loop in a docker container.
 
 I'm facing two issues at the moment:

1.  big data usage, proxy is clearing 10GB after one day of once docker instance
2. after running about one day I get error "filedescriptor out of range in select()"

I figured out that adding user_data_dir to sb solves problem of data usage, but I'm not able to mount volume to my docker container (either mount volume or use docker managed one) so it can persist because chrome driver won't start with error:

selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:57455
from chrome not reachable

(starting docker container as sudo, folder had all required permissions, other volumes are working fine)
